lib/datadog/tracing/contrib/aws/configuration/settings.rb

```diff
@@ -13,21 +13,29 @@ module Configuration
           # @public_api
           class Settings < Contrib::Configuration::Settings
             option :enabled do |o|
-              o.default { env_to_bool(Ext::ENV_ENABLED, true) }
+              o.type :bool
+              o.env_var Ext::ENV_ENABLED
+              o.default true
             end
 
             option :analytics_enabled do |o|
-              o.default { env_to_bool(Ext::ENV_ANALYTICS_ENABLED, false) }
+              o.type :bool
+              o.env_var Ext::ENV_ANALYTICS_ENABLED
+              o.default false
             end
 
             option :analytics_sample_rate do |o|
-              o.default { env_to_float(Ext::ENV_ANALYTICS_SAMPLE_RATE, 1.0) }
+              o.type :float
+              o.env_var Ext::ENV_ANALYTICS_SAMPLE_RATE
+              o.default 1.0
             end
 
             option :service_name do |o|
-              o.default do
+              o.type :string, nil: true
+              o.env_var Ext::ENV_SERVICE_NAME
+              o.setter do |value|
                 Contrib::SpanAttributeSchema.fetch_service_name(
-                  Ext::ENV_SERVICE_NAME,
+                  value,
                   Ext::DEFAULT_PEER_SERVICE_NAME
                 )
               end
```

Comment:
> Also: why use o.setter and instead of o.default?

We use setter rather than default because:
- We changed the `Contrib::SpanAttributeSchema.fetch_service_name` to not check ENV directly, but just values provided to it. 
- The `default` block is not executed with the customer setting value. 

So here is the setter `value` could be the ENV variable `Ext::ENV_SERVICE_NAME` or what ever the customer configured via `Datadog.configure` block